[Hub Generated] Review request for Microsoft.Insights to add version stable/2016-06-01

ToddKingMSFT opened the pull request:
This is a PR generated at [OpenAPI Hub](https://aka.ms/openapihub). You can view your [work branch via this link](https://portal.azure-devex-tools.com/branch/ToddKingMSFT/azure-rest-api-specs/dev-monitor-Microsoft.Insights-2016-06-01...master/).


### Changelog
Please ensure to add changelog with this PR by answering the following questions.
  1. What's the purpose of the update?    
      - [ ] new service onboarding 
      - [x] new API version 
      - [ ] update existing version for new feature 
      - [x] update existing version to fix swagger quality issue in s360
      - [ ] Other, please clarify 
  2. When you are targeting to deploy new service/feature to public regions? Please provide date, or month to public if date is not available yet.
  3. When you expect to publish swagger? Please provide date, or month to public if date is not available yet.
  4. If it's an update to existing version,  please select SDKs of specific language and CLIs that require refresh after swagger is published.
      - [ ] SDK of .NET (need service team to ensure code readiness)
      - [ ] SDK of Python
      - [ ] SDK of Java
      - [ ] SDK of Js
      - [ ] SDK of Go
      - [ ] PowerShell
      - [ ] CLI
      - [ ] Terraform
      - [ ] No, no need to refresh for updates in this PR

### Contribution checklist:
- [x] I commit to follow the [Breaking Change Policy](http://aka.ms/bcforapi) of "no breaking changes"
- [x] I have reviewed the [documentation](https://aka.ms/ameonboard) for the workflow.
- [x] [Validation tools](https://aka.ms/swaggertools) were run on swagger spec(s) and errors have all been fixed in this PR. [How to fix?](https://aka.ms/ci-fix)

If any further question about AME onboarding or validation tools, please view the [FAQ](https://aka.ms/faqinprreview).

### ARM API Review Checklist
- [ ] Ensure to check this box if one of the following scenarios meet updates in the PR, so that label “WaitForARMFeedback” will be added automatically to involve ARM API Review. Failure to comply may result in delays for manifest application. Note this does not apply to data plane APIs, all “removals” and “adding a new property” no more require ARM API review.
  - Adding new API(s)
  - Adding a new API version
  -  [ ] Ensure to copy the existing version into new directory structure for first commit (including refactoring) and then push new changes including version updates in separate commits. This is required to review the changes efficiently.
  - Adding a new service

- [x] Please ensure you've reviewed following [guidelines](https://aka.ms/rpguidelines) including [ARM resource provider contract](https://github.com/Azure/azure-resource-manager-rpc) and [REST guidelines](https://github.com/microsoft/api-guidelines/blob/vNext/azure/Guidelines.md). Estimated time (4 hours). This is required before you can request review from ARM API Review board.

- [ ] If you are blocked on ARM review and want to get the PR merged with urgency, please get the ARM oncall for reviews (*RP Manifest Approvers* team under <ins>Azure Resource Manager service</ins>) from IcM and reach out to them. 

### Breaking Change Review Checklist 
If there are following updates in the PR, ensure to request an approval from Breaking Change Review Board as defined in the [Breaking Change Policy](http://aka.ms/bcforapi). 

- [ ] Removing API(s) in stable version
- [ ] Removing properties in stable version
- [ ] Removing API version(s) in stable version
- [ ] Updating API in stable or public preview version with Breaking Change Validation errors
- [ ] Updating API(s) in public preview over 1 year (refer to [Retirement of Previews](https://dev.azure.com/msazure/AzureWiki/_wiki/wikis/AzureWiki.wiki/37683/Retirement-of-Previews))

**Action**: to initiate an evaluation of the breaking change, create a new intake using the [template for breaking changes](https://aka.ms/Breakingchangetemplate). Addition details on the process and office hours are on the [Breaking change Wiki](https://dev.azure.com/msazure/AzureWiki/_wiki/wikis/AzureWiki.wiki/37684/Breaking-Changes).

Please follow the link to find more details on [PR review process](https://aka.ms/SwaggerPRReview).

ToddKingMSFT committed:
Adds base for updating Microsoft.Insights from version stable/2016-09-01 to version 2016-06-01

ToddKingMSFT committed:
Updates readme

ToddKingMSFT committed:
Updates API version in new specs and examples

ToddKingMSFT committed:
Delete serviceDiagnosticsSettings_API.json

ToddKingMSFT committed:
Delete createOrUpdateServiceDiagnosticSetting.json

ToddKingMSFT committed:
Delete getServiceDiagnosticSetting.json

ToddKingMSFT committed:
Delete updateServiceDiagnosticSetting.json

ToddKingMSFT committed:
Delete GetMetricFiltered.json

ToddKingMSFT committed:
Update GetMetric.json

ToddKingMSFT committed:
Update metrics_API.json

ToddKingMSFT committed:
Update readme.md

ToddKingMSFT commented:
/azp run unifiedPipeline

kairu-ms commented:
/azp run

kairu-ms commented on specification/monitor/resource-manager/readme.md:
```diff
@@ -27,16 +27,26 @@ These are the global settings for the MonitorClient API.
 description: Monitor Management Client
 openapi-type: arm
-tag: package-2020-03
+tag: package-2016-06
```
```suggestion
tag: package-2020-03
```

ToddKingMSFT replied:
Thanks for reviewing, I've incorporated your suggestion.

kairu-ms commented on specification/monitor/resource-manager/readme.md:
````diff
@@ -27,16 +27,26 @@ These are the global settings for the MonitorClient API.
 ```
 
+
+### Tag: package-2016-06
+
+These settings apply only when `--tag=package-2016-06` is specified on the command line.
+
+```yaml $(tag) == 'package-2016-06'
````
```suggestion
### Tag: package-2016-06-only

These settings apply only when `--tag=package-2016-06-only` is specified on the command line.

```yaml $(tag) == 'package-2016-06-only'
```

ToddKingMSFT replied:
Thanks for reviewing, I've incorporated your suggestion.

kairu-ms commented on specification/monitor/resource-manager/Microsoft.Insights/stable/2016-06-01/metrics_API.json:
```diff
@@ -0,0 +1,265 @@
+                        "timeStamp": "2016-11-23T19:16:00Z"
+                      },
+                      {
+                        "timeStamp": "2016-11-23T19:17:00Z"
+                      },
+                      {
+                        "timeStamp": "2016-11-23T19:18:00Z"
+                      },
+                      {
+                        "timeStamp": "2016-11-23T19:19:00Z",
+                        "total": 1.0
+                      },
+                      {
+                        "timeStamp": "2016-11-23T19:20:00Z"
+                      },
+                      {
+                        "timeStamp": "2016-11-23T19:21:00Z"
+                      },
+                      {
+                        "timeStamp": "2016-11-23T19:22:00Z"
+                      },
+                      {
+                        "timeStamp": "2016-11-23T19:23:00Z"
+                      },
+                      {
+                        "timeStamp": "2016-11-23T19:24:00Z",
+                        "total": 1.0
+                      }
+                    ],
+                    "id": "/subscriptions/df602c9c-7aa0-407d-a6fb-eb20c8bd1192/resourceGroups/andy0802australia/providers/Microsoft.Logic/workflows/andy0803australia/providers/microsoft.insights/metrics/RunsStarted",
+                    "name": {
+                      "value": "RunsStarted",
+                      "localizedValue": "Runs Started"
+                    },
+                    "type": "Microsoft.Insights/metrics",
+                    "unit": "2"
+                  }
+                ]
+              }
+            }
```
examples usually link to the example file, For example:
https://github.com/Azure/azure-rest-api-specs/blob/f1d8c4ebb54fdc1301eb2f2a465982cf58b54f69/specification/network/resource-manager/Microsoft.Network/stable/2020-11-01/customIpPrefix.json#L142-L147

ToddKingMSFT replied:
Okay I've removed the inline example in favor of the existing example file already in the examples directory

kairu-ms commented:
Please fix ci issues.

ToddKingMSFT committed:
Update metrics_API.json

Update casing of Microsoft.Insights

ToddKingMSFT committed:
Update metrics_API.json

Incorporate PR feedback to remove inline example and use common error response definition

ToddKingMSFT committed:
Update GetMetric.json

Fix formatting errors

ToddKingMSFT committed:
Update readme.md

Incorporate PR feedback

ToddKingMSFT committed:
Create operations_API.json

Add operations API

ToddKingMSFT committed:
Create OperationList.json

ToddKingMSFT committed:
Update readme.md

Add operations API

ToddKingMSFT committed:
Update readme.md

Add suppressions for error response format due to it being a breaking change.

ToddKingMSFT committed:
Update operations_API.json

Fix spelling mistake

ToddKingMSFT commented:
@kairu-ms Can I get some help completing this PR?

kairu-ms commented:
This PR LGTM. @akning-ms Could you please merge this PR? Thanks.

ToddKingMSFT commented:
@akning-ms Could you help me merge this PR?

ToddKingMSFT committed:
Update metrics_API.json

Go back to common ErrorResponse format for Azure Monitor

ToddKingMSFT committed:
Merge branch 'Azure:master' into dev-monitor-Microsoft.Insights-2016-06-01

ToddKingMSFT committed:
Update metrics_API.json

Address new lint errors

ToddKingMSFT committed:
Update operations_API.json

Fix new lint errors

ToddKingMSFT commented:
Hello @akning-ms / @kairu-ms I've fixed the new lint violations that appeared. The "SDK azure-sdk-for-net" failure is a known issue that has been failing since last last year, I have a separate thread with the DataCollectionRules team on fixing that, I had a review with JeffreyRichter on the "Swagger SDK Breaking Change Tracking" failure and he approved it.
I don't have permissions to complete this PR so I need help from someone to complete this. If that's not you can you point me at who I should ask?
